service/controller/chart/resource/release/resource.go

```diff
@@ -183,6 +183,29 @@ func isEmpty(c ReleaseState) bool {
 	return equals(c, ReleaseState{})
 }
 
+func isReleaseFailed(current, desired ReleaseState) bool {
```

Comment:
When I initially read this, I had to double back because of the switch between `true` and `false` in the conditionals. Maybe this would be more readable:

```go
func isReleaseFailed(current, desired ReleaseState) bool {
        retVal := false
        
	if ! isEmpty(current) {
	  
          // Values have changed so we will try to update the release.
	  if current.ValuesMD5Checksum == desired.ValuesMD5Checksum {
	  	  retVal = true
	  }

	  // Version has changed so we will try to update the release.
	  if current.Version == desired.Version {
		  retVal = true
	  }

	  // Release is failed and should not be updated.
	  if current.Status == helmFailedStatus {
		  retVal = true
	  }
	
	}

	return retVal
}
```

To make it terser, could just make it one just logical OR conditional... But I think the multiple `if` probably makes it more readable.

Reply:
@yasn77 Go is more verbose than other languages but this is a pretty common pattern.

The problem is this is based on `isReleaseModified` and I don't want to change that as well. With the helm3 branch and flattening the operator I don't want to change more that is required.

@tomahawk28 WDYT?

Reply:
flapping `true` or `false` is a bit hard to read, of course, @yasn77 is right at this point, maybe we can do it on next PR.

Reply:
oh, no, this is a new function, so changing doesn't matter so far.

Reply:
I'll do a follow up and also update isReleaseModified to keep them in sync. I want to get this merged so I can cut a release. As its needed for the KVM 9.0.1 release that is being prepped.

Thanks both for the feedback. ❤️